packages/gestalt/src/Table.js

```diff
@@ -10,23 +11,61 @@ import TableHeaderCell from './TableHeaderCell.js';
 import TableRowExpandable from './TableRowExpandable.js';
 import TableRow from './TableRow.js';
 import TableSortableHeaderCell from './TableSortableHeaderCell.js';
+import TableContext from './TableContextProvider.js';
 
 type Props = {|
   children: Node,
   borderStyle?: 'sm' | 'none',
   maxHeight?: number | string,
+  stickyColumns?: number,
 |};
 
 export default function Table(props: Props): Node {
-  const { borderStyle, children, maxHeight } = props;
+  const { borderStyle, children, maxHeight, stickyColumns } = props;
+  const [showShadowScroll, setShowShadowScroll] = useState(null);
+  const contentRef = useRef<?HTMLElement>(null);
 
+  const updateShadows = useCallback(() => {
+    const target = contentRef.current;
+    if (!target) {
+      return;
+    }
+    if (target.scrollLeft > 0) {
+      setShowShadowScroll('right');
+    } else if (target.scrollLeft < 0) {
+      setShowShadowScroll('left');
+    } else {
+      setShowShadowScroll(null);
+    }
+  }, []);
```

Comment:
This seems like it might need `contentRef` as a dependency? It looks like it could have a stale value in the closure.

Reply:
I think refs aren't required. eslintreact-hooks/exhaustive-deps would complain otherwise.

Reply:
![image](https://user-images.githubusercontent.com/10593890/113351724-406ff580-9309-11eb-9b45-7261e13a62bd.png)